tests/unittests/CoreGraphics.drawing/CGContextDrawing_WidenPathTests.cpp

```diff
@@ -0,0 +1,205 @@
+    CGPathAddLineToPoint(thepath, NULL, xstart + .1 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .3 * width, ystart + .5 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .25 * width, ystart + .9 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .5 * width, ystart + .7 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .75 * width, ystart + .9 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .7 * width, ystart + .5 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .9 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .65 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .5 * width, ystart + .1 * height);
+    CGPathCloseSubpath(thepath);
+
+    CGLineCap lineCap = ::testing::get<0>(GetParam());
+    CGLineJoin lineJoin = ::testing::get<1>(GetParam());
+    CGFloat lineWidth = ::testing::get<2>(GetParam());
+    CGFloat miterLimit = ::testing::get<3>(GetParam());
+    bool useContext = ::testing::get<4>(GetParam());
+
+    drawStrokeTestWithParams(context, thepath, lineCap, lineJoin, lineWidth, miterLimit, useContext);
+
+    CGPathRelease(thepath);
+}
+
+DRAW_TEST_P(CGPathCopyByStroking_Curves, PathWiden) {
+    CGContextRef context = GetDrawingContext();
+    CGRect drawingBounds = GetDrawingBounds();
+    CGFloat width = drawingBounds.size.width;
+    CGFloat height = drawingBounds.size.height;
+    CGFloat xstart = drawingBounds.origin.x;
+    CGFloat ystart = drawingBounds.origin.y;
+
+    CGAffineTransform transformation = CGAffineTransformIdentity;
+    transformation = CGAffineTransformScale(transformation, .8, .8);
+    transformation = CGAffineTransformTranslate(transformation, .1 * width, .1 * height);
+
+    CGMutablePathRef thepath = CGPathCreateMutable();
+    CGPathMoveToPoint(thepath, &transformation, xstart + .75 * width, ystart + .5 * height);
+    CGPathAddArc(thepath, &transformation, xstart + .5 * width, ystart + .5 * height, .5 * height, 0, M_PI / 2, true);
+    CGPathAddArc(thepath, &transformation, xstart + .5 * width, ystart + .5 * height, .5 * height, M_PI / 2, 0, true);
+
+    CGLineCap lineCap = ::testing::get<0>(GetParam());
+    CGLineJoin lineJoin = ::testing::get<1>(GetParam());
+    CGFloat lineWidth = ::testing::get<2>(GetParam());
+    CGFloat miterLimit = ::testing::get<3>(GetParam());
+    bool useContext = ::testing::get<4>(GetParam());
+
+    drawStrokeTestWithParams(context, thepath, lineCap, lineJoin, lineWidth, miterLimit, useContext);
+
+    CGPathRelease(thepath);
+}
+
+INSTANTIATE_TEST_CASE_P(CGPathWidenTest,
+                        CGPathCopyByStroking_Lines,
+                        ::testing::Combine(::testing::ValuesIn(lineCaps),
+                                           ::testing::ValuesIn(lineJoins),
+                                           ::testing::ValuesIn(lineWidths),
```

Comment:
we might want to test the zero-width cases separately so we don't spit out a hundred identical test images with different names.